src/lib/tradeExecution.ts

```diff
@@ -93,13 +95,19 @@ export class TradeExecution {
       // Given the intersection of the inherent bits of sc wallets (only one chain, not deployed on others) and EVM chains (same address on every chain)
       // this means that this is absolutely fine, as in case of multi-hops, the first hop and the last would be the same addy
       const accountId = selectFirstHopSellAccountId(store.getState())
+
+      const swap = selectSwapByQuoteId(store.getState(), {
+        quoteId: tradeQuote.id as unknown as QuoteId,
```

Comment:
That looks off, why do we have a nominal then? Also the types don't overlap here i.e string vs. number
<img width="1216" alt="image" src="https://github.com/user-attachments/assets/6253cf8a-1d16-4268-9ede-4932bfefb5ae" />